`vue-tsc --noEmit` does not support `setup enum` of vue 3.2-beta

### Describe the bug

[`template-vue-ts`](https://github.com/vitejs/vite/blob/main/packages/create-vite/template-vue-ts/package.json#L17) relies on a low version of [`vue-tsc`](https://www.npmjs.com/package/vue-tsc), and does not support the [`setup enum`](https://github.com/vuejs/vue-next/issues/3571) of vue 3.2-beta.

This issue can be solved by upgrading the `vue-tsc` version to `0.2.2`, and I will try to provide a PR later.

### Reproduction

This [example](https://github.com/lijialiang/vite-vue-ts-tsc-issue) of my issue.

```shell
$ yarn install

$ yarn build
```

```shell
# error info
$ vue-tsc --noEmit && vite build
src/components/HelloWorld.vue:2:22 - error TS2552: Cannot find name 'Fruit'. Did you mean 'fruit'?

2   <p v-if="fruit === Fruit.apple">
                       ~~~~~


src/components/HelloWorld.vue:5:22 - error TS2552: Cannot find name 'Fruit'. Did you mean 'fruit'?

5   <p v-if="fruit === Fruit.banana">
                       ~~~~~



Found 2 errors.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### System Info

```shell
System:
    OS: macOS 11.2.1
    CPU: (8) x64 Apple M1
    Memory: 59.19 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.1 - ~/.nvm/versions/node/v14.15.1/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.15.1/bin/npm
  Browsers:
    Chrome: 91.0.4472.164
    Safari: 14.0.3
  npmPackages:
    @vitejs/plugin-vue: ^1.2.5 => 1.2.5 
    vite: ^2.4.3 => 2.4.3
```

### Used Package Manager

yarn

### Logs

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/.github/contributing.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/vue-next instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).

--- a/packages/create-vite/template-vue-ts/package.json
+++ b/packages/create-vite/template-vue-ts/package.json
@@ -14,6 +14,6 @@
     "@vue/compiler-sfc": "^3.0.5",
     "typescript": "^4.3.2",
     "vite": "^2.4.3",
-    "vue-tsc": "^0.0.24"
+    "vue-tsc": "^0.2.2"
   }
 }

